### ISSUE TITLE
BUG: limitation in stats.binom.ppf()

### ISSUE THREAD

**Issue by lenzone91** [scipy/scipy#19518]
### Describe your issue.

Hi, 
I discovered a limitation in the "stats.binom.ppf(q,n,p)" function.
When the probability 'p' is set to 1  and 0<'q'=<1,  i expect the result to be 'n'.
However this is not the case when 'q' <= 0.5.

This limitation didn't exist before boost implementation in v=1.7.0. 

### Reproducing Code Example

```python
from scipy.stats import binom

print(binom.ppf(0.49, 50, 1)) 
# With scipy >= 1.7.0 result = 0.0
# With scipy < 1.7.0 result = 50.0

print(binom.ppf(0.51, 50, 1))
# With scipy >= 1.7.0 result = 50.0
# With scipy < 1.7.0 result = 50.0
```

### Error message

```shell
None
```

### SciPy/NumPy/Python version and system information

```shell
New version with limitation :
1.10.1 1.24.4 sys.version_info(major=3, minor=8, micro=17, releaselevel='final', serial=0)
/home/c17180/micromamba/envs/test_alm/lib/python3.8/site-packages/scipy/__config__.py:140: UserWarning: Install `pyyaml` for better output
  warnings.warn("Install `pyyaml` for better output", stacklevel=1)
{
  "Compilers": {
    "c": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.2.1",
      "commands": "cc"
    },
    "cython": {
      "name": "cython",
      "linker": "cython",
      "version": "0.29.33",
      "commands": "cython"
    },
    "c++": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.2.1",
      "commands": "c++"
    },
    "fortran": {
      "name": "gcc",
      "linker": "ld.bfd",
      "version": "10.2.1",
      "commands": "gfortran"
    },
    "pythran": {
      "version": "0.12.1",
      "include directory": "/tmp/pip-build-env-q2fwe5jt/overlay/lib/python3.8/site-packages/pythran"
    }
  },
  "Machine Information": {
    "host": {
      "cpu": "x86_64",
      "family": "x86_64",
      "endian": "little",
      "system": "linux"
    },
    "build": {
      "cpu": "x86_64",
      "family": "x86_64",
      "endian": "little",
      "system": "linux"
    },
    "cross-compiled": false
  },
  "Build Dependencies": {
    "blas": {
      "name": "OpenBLAS",
      "found": true,
      "version": "0.3.18",
      "detection method": "cmake",
      "include directory": "unknown",
      "lib directory": "unknown",
      "openblas configuration": "unknown",
      "pc file directory": "unknown"
    },
    "lapack": {
      "name": "OpenBLAS",
      "found": true,
      "version": "0.3.18",
      "detection method": "cmake",
      "include directory": "unknown",
      "lib directory": "unknown",
      "openblas configuration": "unknown",
      "pc file directory": "unknown"
    }
  },
  "Python Information": {
    "path": "/opt/python/cp38-cp38/bin/python",
    "version": "3.8"
  }
}

Old version without limitation :
1.6.3 1.22.4 sys.version_info(major=3, minor=8, micro=17, releaselevel='final', serial=0)
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
```

**Comment by mdhaber** [comment]
I can't reproduce this on my Mac or Windows machines or on Colab. Is anyone else seeing this?

**Comment by tylerjereddy** [comment]
I tested on a third platform (x86_64 Linux) on latest `main` and also cannot reproduce the problem--I get same result for both:

```
50.0
50.0
```

**Comment by mdhaber** [comment]
@lenzone91 Can you try installing the latest version of SciPy and letting us know if the problem persists? Can you reproduce it on another machine?

**Comment by lenzone91** [comment]
After some tests, the error was fixed in version 1.11.0. 
Thank you all !
